### ISSUE TITLE
Bug: Forced HTTPS wrong redirect

### ISSUE THREAD

**Issue by WaaaghNL** [codeigniter4/CodeIgniter4#3260]
**Describe the bug**
When `$forceGlobalSecureRequests == True;` then the redirect creates a strange URI scheme http://ci.domain.com/ redirects to https://https//ci.domain.com// and it adds an extra tailing / to the end of an url if you add the / on the end of .com

**CodeIgniter 4 version**
Latest version that composer is pulling down with `composer create-project codeigniter4/appstarter project-root`

**Comment by michalsn** [comment]
I was able to recreate the issue with the extra trailing slash, but not with "extra" `https//`. 

* Can you tell us what server are you using? 
* Are there any changes made to the `.htaccess` file? 
* What is your `baseURL`?

As I understand, if you try to enter `http://ci.domain.com/` address then you're redirected to the `https://https//ci.domain.com//`?

**Comment by WaaaghNL** [comment]
**Server info**
```
cPanel Version | 88.0 (build 12)
Apache Version | 2.4.43
PHP Version | 7.3.19
MySQL Version | 10.2.32-MariaDB
Architecture | x86_64
Operating System | linux
```

**Changes to HTACCESS**
None, just a clean download using the latest composer version. no addons like auth etc.

**Changes to App.php**
public $baseURL = 'https://ci.domain.nl/';
public $appTimezone = 'Europe/Amsterdam';
public $forceGlobalSecureRequests = false;

**Comment by michalsn** [comment]
> ```public $forceGlobalSecureRequests = false;```

Isn't it set to `true`?

**Comment by WaaaghNL** [comment]
Correct, switched it back so i can start to build something

**Comment by michalsn** [comment]
Okay, thanks.

**Comment by samsonasik** [comment]
should can be closed in favor of #3292
